Adding AWS Elasticsearch representation

haverma opened the pull request:
1.) Similar to RDS instance representation implementation

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/batfish/batfish/880)
<!-- Reviewable:end -->

arifogel commented:
Mostly good. Please move your test configurations as suggested and address comments.

---

Reviewed 26 of 26 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java, line 31 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48HR3vOcg80fPjegRx:-L48HR3vOcg80fPjegRy:b-70ndhq) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java#L31)):*
> ```Java
>   private List<String> _subnets;
> 
>   private transient IpAccessList _inAcl;
> ```

Why is this a field instead of a local variable?

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java, line 33 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48HVUx8oKZsU_GAVT3:-L48HVUx8oKZsU_GAVT4:b-70ndhq) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java#L33)):*
> ```Java
>   private transient IpAccessList _inAcl;
> 
>   private transient IpAccessList _outAcl;
> ```

Why is this a field instead of a local variable?

---

*[projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java, line 71 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48IAQw_pHfEwa7J1p4:-L48IAQw_pHfEwa7J1p5:bqv33qx) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java#L71)):*
> ```Java
>         BatfishTestUtils.getBatfishFromTestrigText(
>             TestrigText.builder()
>                 .setAwsText("org/batfish/representation/aws/testrigs", awsFilenames)
> ```

I'm confused. Which testrig is this?

---

*[projects/batfish/src/test/java/org/batfish/representation/aws/RdsInstanceTest.java, line 76 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48JLH9Ylz7ai1kCDSI:-L48JLHARxEyZKdmccu4:b15qvjp) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/test/java/org/batfish/representation/aws/RdsInstanceTest.java#L76)):*
> ```Java
>         BatfishTestUtils.getBatfishFromTestrigText(
>             TestrigText.builder()
>                 .setAwsText("org/batfish/representation/aws/testrigs", awsFilenames)
> ```

Again, what testrig is this? Is this a testrig named `testrigs`? If so, the files should be one level deeper in an appropriately named testrig.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/880#-:-L48K7NaTO6mmu3jKpVD:bp9snop)*
<!-- Sent from Reviewable.io -->

haverma commented:
Renamed and addressed

---

Review status: 8 of 20 files reviewed at latest revision, 4 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java, line 31 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48HR3vOcg80fPjegRx:-L48tnW0YpiJcWWUuR_t:b-n9tkhd) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java#L31)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Why is this a field instead of a local variable?
</blockquote></details>

changed

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java, line 33 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48HVUx8oKZsU_GAVT3:-L48tMR4h3xjJM-4cBYF:b-n9tkhd) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/main/java/org/batfish/representation/aws/ElasticsearchDomain.java#L33)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Why is this a field instead of a local variable?
</blockquote></details>

changed

---

*[projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java, line 71 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48IAQw_pHfEwa7J1p4:-L48tOIjd7o45TORDAEX:bhfojyj) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java#L71)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

I'm confused. Which testrig is this?
</blockquote></details>

it is a test testrig being used for rds and elasticsearch tests currently, changed to just "test"

---

*[projects/batfish/src/test/java/org/batfish/representation/aws/RdsInstanceTest.java, line 76 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48JLH9Ylz7ai1kCDSI:-L48ta_iFPagz745cbbg:b-89yzof) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/test/java/org/batfish/representation/aws/RdsInstanceTest.java#L76)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Again, what testrig is this? Is this a testrig named `testrigs`? If so, the files should be one level deeper in an appropriately named testrig.
</blockquote></details>

addressed as above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/880#-:-L48tk-0M-wE1UO-7H4g:b8xg2q3)*
<!-- Sent from Reviewable.io -->

arifogel commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 7 of 34 files at r1, 20 of 20 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java, line 71 at r1](https://reviewable.io:443/reviews/batfish/batfish/880#-L48IAQw_pHfEwa7J1p4:-L4D36auzXwyV8DFvN55:b-hcocle) ([raw file](https://github.com/batfish/batfish/blob/18d859574b1c79d34b4c9df729cdb82e2928bd58/projects/batfish/src/test/java/org/batfish/representation/aws/ElasticsearchDomainTest.java#L71)):*
<details><summary><i>Previously, haverma (Harsh Verma) wrote…</i></summary><blockquote>

it is a test testrig being used for rds and elasticsearch tests currently, changed to just "test"
</blockquote></details>

Good enough for now. Later if you need more distinct aws testrigs for testing, change this to: `org/batfish/representation/aws/testrigs/<something>` so there is a natural place for other test aws testrigs to go.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/880#-:-L4D3sxdvqlONIgA7v9s:bnfp4nl)*
<!-- Sent from Reviewable.io -->
